mycelo/genesis/genesis_state.go

```diff
@@ -608,87 +676,303 @@ func (ctx *deployContext) deployReserve() error {
+	validators := ctx.contract("Validators")
+
+	groupRequiredGold := new(big.Int).Mul(
+		ctx.genesisConfig.Validators.GroupLockedGoldRequirements.Value,
+		big.NewInt(int64(ctx.accounts.ValidatorsPerGroup)),
+	)
+	groupComission := ctx.genesisConfig.Validators.Commission.BigInt()
+
+	for _, group := range validatorGroupsAccounts {
+		address := group.Address
+		logger := ctx.logger.New("group", address)
+
+		ctx.statedb.AddBalance(address, groupRequiredGold)
+
+		logger.Info("Lock group gold", "amount", groupRequiredGold)
+		if _, err := lockedGold.Call(contract.CallOpts{Origin: address, Value: groupRequiredGold}, "lock"); err != nil {
+			return err
 		}
 
-		if !authorized {
-			ctx.logger.Warn("Fixing StableToken goldprice requires setting admin as oracle", "admin", ctx.adminAccount.Address)
-			err = ctx.contract("SortedOracles").SimpleCall("addOracle", stableTokenAddress, ctx.adminAccount.Address)
-			if err != nil {
+		logger.Info("Register group")
+		if err := validators.SimpleCallFrom(address, "registerValidatorGroup", groupComission); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ctx *deployContext) addValidatorsToGroups() error {
+	validators := ctx.contract("Validators")
+
+	validatorGroups := ctx.accounts.ValidatorGroups()
+	for groupIdx, group := range validatorGroups {
+		groupAddress := group.Address
+		prevGroupAddress := common.ZeroAddress
+		if groupIdx > 0 {
+			prevGroupAddress = validatorGroups[groupIdx-1].Address
+		}
+
+		for i, validator := range group.Validators {
+			ctx.logger.Info("Add validator to group", "validator", validator.Address, "group", groupAddress)
+
+			// affiliate validators to group
+			if err := validators.SimpleCallFrom(validator.Address, "affiliate", groupAddress); err != nil {
 				return err
 			}
+
+			// accept validator as group member
+			if i == 0 {
+				if err := validators.SimpleCallFrom(groupAddress, "addFirstMember", validator.Address, common.ZeroAddress, prevGroupAddress); err != nil {
+					return err
+				}
+			} else {
+				if err := validators.SimpleCallFrom(groupAddress, "addMember", validator.Address); err != nil {
+					return err
+				}
+			}
 		}
+	}
 
-		ctx.logger.Info("Reporting price of StableToken to oracle")
-		err = ctx.contract("SortedOracles").SimpleCall("report",
-			stableTokenAddress,
-			ctx.GenesisConfig.StableToken.GoldPrice.BigInt(),
-			common.ZeroAddress,
-			common.ZeroAddress,
-		)
-		if err != nil {
+	return nil
+}
+
+func (ctx *deployContext) voteForGroups() error {
+	election := ctx.contract("Election")
+
+	validatorGroups := ctx.accounts.ValidatorGroupAccounts()
+
+	// value previously locked on registerValidatorGroups()
+	lockedGoldOnGroup := new(big.Int).Mul(
+		ctx.genesisConfig.Validators.GroupLockedGoldRequirements.Value,
+		big.NewInt(int64(ctx.accounts.ValidatorsPerGroup)),
+	)
+
+	// current group order (see `addFirstMember` on addValidatorsToGroup) is:
+	// [ 1stGroup, 2ndGroup, ..., lastgroup]
```

Comment:
```suggestion
	// [ groupZero, groupOne, ..., lastGroup]
```
(see below for more)